core/scripts/chaincli/handler/debug.go

```diff
@@ -229,73 +236,75 @@ func (k *Keeper) Debug(ctx context.Context, args []string) {
 			message(fmt.Sprintf("StreamsLookup data: {FeedParamKey: %s, Feeds: %v, TimeParamKey: %s, Time: %d, ExtraData: %s}", streamsLookupErr.FeedParamKey, streamsLookupErr.Feeds, streamsLookupErr.TimeParamKey, streamsLookupErr.Time.Uint64(), hexutil.Encode(streamsLookupErr.ExtraData)))
-			if streamsLookupErr.FeedParamKey == feedIdHex && streamsLookupErr.TimeParamKey == blockNumber {
-				message("using mercury lookup v0.2")
-				// handle v0.2
-				cfg, err := keeperRegistry21.GetUpkeepPrivilegeConfig(triggerCallOpts, upkeepID)
-				if err != nil {
-					failUnknown("failed to get upkeep privilege config ", err)
-				}
-				allowed := false
-				if len(cfg) > 0 {
-					var privilegeConfig streams.UpkeepPrivilegeConfig
-					if err := json.Unmarshal(cfg, &privilegeConfig); err != nil {
-						failUnknown("failed to unmarshal privilege config ", err)
-					}
-					allowed = privilegeConfig.MercuryEnabled
-				}
-				if !allowed {
-					resolveIneligible("upkeep reverted with StreamsLookup but is not allowed to access streams")
-				}
-			} else if streamsLookupErr.FeedParamKey != feedIDs || streamsLookupErr.TimeParamKey != timestamp {
-				// handle v0.3
-				resolveIneligible("upkeep reverted with StreamsLookup but the configuration is invalid")
-			} else {
-				message("using mercury lookup v0.3")
+
+			// check if upkeep is allowed to use mercury
+			_, _, _, allowed, err := streams.AllowedToUseMercury(latestCallOpts, upkeepID)
+			if err != nil {
+				failUnknown("failed to check if upkeep is allowed to use mercury", err)
+			}
+			if !allowed {
+				resolveIneligible("upkeep reverted with StreamsLookup but is not allowed to access streams")
+			}
+
+			streamsLookup := &mercury.StreamsLookup{
+				StreamsLookupError: &mercury.StreamsLookupError{
+					FeedParamKey: streamsLookupErr.FeedParamKey,
+					Feeds:        streamsLookupErr.Feeds,
+					TimeParamKey: streamsLookupErr.TimeParamKey,
+					Time:         streamsLookupErr.Time,
+					ExtraData:    streamsLookupErr.ExtraData,
+				},
+				UpkeepId: upkeepID,
+				Block:    blockNum,
 			}
-			streamsLookup := &StreamsLookup{streamsLookupErr.FeedParamKey, streamsLookupErr.Feeds, streamsLookupErr.TimeParamKey, streamsLookupErr.Time, streamsLookupErr.ExtraData, upkeepID, blockNum}
 
 			if k.cfg.MercuryLegacyURL == "" || k.cfg.MercuryURL == "" || k.cfg.MercuryID == "" || k.cfg.MercuryKey == "" {
 				failCheckConfig("Mercury configs not set properly, check your MERCURY_LEGACY_URL, MERCURY_URL, MERCURY_ID and MERCURY_KEY", nil)
 			}
-			handler := NewMercuryLookupHandler(&MercuryCredentials{k.cfg.MercuryLegacyURL, k.cfg.MercuryURL, k.cfg.MercuryID, k.cfg.MercuryKey}, k.rpcClient)
-			state, failureReason, values, _, err := handler.doMercuryRequest(ctx, streamsLookup)
-			if failureReason == UpkeepFailureReasonInvalidRevertDataInput {
+
+			// do mercury request
+			ocr2KeepersCheckResult := mustOcr2KeepersCheckResult(upkeepID, checkResult, trigger)
```

Comment:
nitpick, but we're trying to move away from the "ocr2keepers" name. It's just "chainlink-automation" not. Can we rename this to just `checkResult` or `automationCheckResult` or similar?